activerecord/lib/active_record/associations/association.rb

```diff
@@ -210,6 +210,18 @@ def create!(attributes = nil, &block)
         _create_record(attributes, true, &block)
       end
 
+      def violates_strict_loading?
+        return unless find_target?
```

Comment:
I added the `find_target?` check to ensure that the loaded association adheres to the strict loading rule, which is necessary for passing the newly added test, **test_preload_audit_logs_are_strict_loading_with_pluck**. 

Previously, the `violates_strict_loading?` method functioned correctly without this check because `find_target?` was used earlier in the process.

https://github.com/rails/rails/blob/82054e8eb76a72ded0abee83019e60c19f552a8b/activerecord/lib/active_record/associations/collection_association.rb#L269-L271

https://github.com/rails/rails/blob/82054e8eb76a72ded0abee83019e60c19f552a8b/activerecord/lib/active_record/associations/association.rb#L220-L223


I'm not sure whether to open a separate PR for this refactoring since it's a minor change. For now, I've added it as a second commit.